src/server/pmix_server_get.c

```diff
@@ -494,7 +526,7 @@ pmix_status_t pmix_pending_resolve(pmix_nspace_t *nptr, int rank,
             /* run through all the requests to this rank */
             PMIX_LIST_FOREACH(req, &lcd->loc_reqs, pmix_dmdx_request_t) {
                 pmix_status_t rc;
-                rc = _satisfy_request(nptr, rank, req->cbfunc, req->cbdata, NULL);
+                rc = _satisfy_request(nptr, rank, NULL, req->cbfunc, req->cbdata, NULL);
                 if( PMIX_SUCCESS != rc ){
```

Comment:
Yeah, I agree with your analysis. Will have to find a way to mark the request, or figure it out after dmodex returns